pyfr/integrators/std/controllers.py

```diff
@@ -77,7 +77,12 @@ def advance_to(self, t):
 
         while self.tcurr < t:
             # Decide on the time step
-            dt = max(min(t - self.tcurr, self._dt), self.dtmin)
+            if self.tcurr + self._dt < t < self.tcurr + 2*self._dt:
```

Comment:
We should be consistent for std and dual controllers about what we do here.

Reply:
I added a function in the base integrator class to be consistent